### DISCUSSION TITLE
Get list network request

### DISCUSSION THREAD

**Discussion by AndrejSayadov** [seleniumbase/SeleniumBase#972] [Q&A] [answered]
Hello! Please tell me how to get a list of requests using python?
<img width="2048" alt="image" src="https://user-images.githubusercontent.com/89129698/130761374-b2103c3b-90b9-440a-ac53-e6326f172198.png">

**Comment by mdmintz** [comment, marked as answer]
@AndrejSayadov Use the following code snippet in your Chrome tests:
```python
items = self.driver.execute_script("var performance = window.performance || window.mozPerformance || window.msPerformance || window.webkitPerformance || {}; var network = performance.getEntries() || {}; return network;")
for item in items:
    print(item)
```

**Reply by suredarv06** [reply]
It is good, returning network request list but how to access the network request's response also?

**Reply by mdmintz** [reply]
Use `wire=True` for Wire Mode, which uses `selenium-wire`. It's not compatible with UC Mode though.

Example:

```python
from seleniumbase import Driver

driver = Driver(wire=True, headless=True)
try:
    driver.get("https://wikipedia.org")
    for request in driver.requests:
        print(request.url)
finally:
    driver.quit()
```

Docs: https://github.com/wkeeling/selenium-wire

**Reply by suredarv06** [reply]
I am using UC mode, where wire option is not supported .  
Thats why  I am going for 
items = self.driver.execute_script("var performance = window.performance || window.mozPerformance || window.msPerformance || window.webkitPerformance || {}; var network = performance.getEntries() || {}; return network;")
for item in items:
    print(item)
**how to access the network request's response also**? ,From this code I am getting response list only.

**Reply by mdmintz** [reply]
Example: [SeleniumBase/examples/uc_cdp_events.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/uc_cdp_events.py)

For more, try StackOverflow.

**Comment by AndrejSayadov** [comment]
This method does not get a response (body) from requests

**Comment by mdmintz** [comment]
@AndrejSayadov You can check through solutions listed in https://stackoverflow.com/questions/5799228/how-to-get-status-code-by-using-selenium-py-python-code
And there might be better Stack Overflow solutions.
